### PR TITLE
PROD-1306: Readiness monitoring base engine

### PR DESCRIPTION
**Pull request by jabdul**
#### Subtask of
- Jira Ticket [PROD-1115](http://jira.photobox.com/browse/PROD-1115)

#### Task
- Jira Ticket [PROD-1306](http://jira.photobox.com/browse/PROD-1306)

#### Depends on release of `ecom-util`...
- `readiness` [PROD-1115](http://jira.photobox.com/browse/PROD-1115)

### PR Tasks

- [x] Use monitoring `readiness` in  `ecom-utils`
- [x] Add `/admin/readiness` route
- [x] Add `dynamodb` check placeholder

### PR Housekeeping

- [ ] Unit Tests
- [ ] Updated Documentation
- [ ] No Linting Errors/ Warnings
- [ ] Passing Build

### CODE REVIEW

File: src/utils/readiness.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  new Promise((resolve, reject) => {
+    setTimeout(reject, maxTimeAllowed, new NotReadyError('Readiness check timeout'));
+  });
+
+const checksExist = checks => {
+  if (!checks.length) throw new NotReadyError('Checks are required.');
+};
+
+const runChecks = async checks =>
+  Promise.all([
```
Comment (hiddentao) [inline src/utils/readiness.js]: This seems unnecessarily complex. Why not just:

```js
const runChecks = checks => Promise.all(checks.map(check =>
  check().catch(err => throw new NotReadyError(e))
))
```

File: src/utils/readiness.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ...checks.map(
+      async check =>
+        await check().catch(e => {
+          throw new NotReadyError(e);
+        })
+    ),
+  ]);
+
+const isReady = async (maxTimeAllowed, completedChecks, logger) =>
```
Comment (hiddentao) [inline src/utils/readiness.js]: I would combine `runChecks` into `isReady`. Also, the use of `Promise.race()` is really cool, but why do you not throw an error if a timeout occurs?

File: src/utils/readiness.js (hunk 1 above)
Comment (jabdul) [reply src/utils/readiness.js]: This code is now in https://github.com/photobox/ecom-utils/pull/17. You may want to add your comment there and I'll respond in kind.

File: src/utils/readiness.js (hunk 2 above)
Comment (jabdul) [reply src/utils/readiness.js]: See my above comment.

**Pull request closed without merging**